test/legacy_test/test_linalg_cholesky_inverse.py

```diff
@@ -0,0 +1,203 @@
+#   Copyright (c) 2024 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+import numpy as np
+
+import paddle
+from paddle import base
+from paddle.base import core
+from paddle.pir_utils import test_with_pir_api as _test_with_pir_api
+
+RTOL = {'float32': 1e-7, 'float64': 1e-11}
```

Comment:
这儿fp64不能是1e-15吗

Reply:
我本地测试 linux 能到 1e-13 ，window 下面精度到 1e-11 ，所以这里写的 1e-11 ～

这个函数精度貌似达不到这么高，对比 torch :

``` python
In [34]: import torch
    ...: import numpy as np
    ...: x = np.array([[3.0, 0.0, 0.0], [2.0, 3.0, 0.0], [-1.0, 1.0, -2.0]])
    ...: A = torch.tensor(x)
    ...: A = A @ A.T + torch.eye(3)*1e-3
    ...: L = torch.linalg.cholesky(A)
    ...: out = torch.cholesky_inverse(L)
    ...: An = x @ x.T + np.eye(3)*1e-3
    ...: on = np.linalg.inv(An)
    ...: np.testing.assert_allclose(out.numpy(), on, rtol=1e-15)
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
Cell In[34], line 10
      8 An = x @ x.T + np.eye(3)*1e-3
      9 on = np.linalg.inv(An)
---> 10 np.testing.assert_allclose(out.numpy(), on, rtol=1e-15)

    [... skipping hidden 1 frame]

File /usr/lib/python3.8/contextlib.py:75, in ContextDecorator.__call__.<locals>.inner(*args, **kwds)
     72 @wraps(func)
     73 def inner(*args, **kwds):
     74     with self._recreate_cm():
---> 75         return func(*args, **kwds)

File ~/venv38dev/lib/python3.8/site-packages/numpy/testing/_private/utils.py:862, in assert_array_compare(comparison, x, y, err_msg, verbose, header, precision, equal_nan, equal_inf, strict)
    858         err_msg += '\n' + '\n'.join(remarks)
    859         msg = build_err_msg([ox, oy], err_msg,
    860                             verbose=verbose, header=header,
    861                             names=('x', 'y'), precision=precision)
--> 862         raise AssertionError(msg)
    863 except ValueError:
    864     import traceback

AssertionError: 
Not equal to tolerance rtol=1e-15, atol=0

Mismatched elements: 9 / 9 (100%)
Max absolute difference: 4.28335145e-12
Max relative difference: 2.79894493e-11
 x: array([[ 0.237564, -0.120313,  0.138811],
       [-0.120313,  0.138848, -0.083284],
       [ 0.138811, -0.083284,  0.249911]])
 y: array([[ 0.237564, -0.120313,  0.138811],
       [-0.120313,  0.138848, -0.083284],
       [ 0.138811, -0.083284,  0.249911]])

```

差不多也是 1e-11 左右 ～

Reply:
在问最后一个问题，torch跟numpy的对比也跟我们比较接近嘛

Reply:
测试用例里面是比对 paddle 与 numpy 的结果 ～

这里单独比对一下 paddle，torch 和 numpy 的结果：

``` python
In [5]: import paddle

In [6]: import torch

In [7]: import numpy as np

In [8]: x = np.array([[3.0, 0.0, 0.0], [2.0, 3.0, 0.0], [-1.0, 1.0, -2.0]])

In [9]: A = torch.tensor(x)

In [10]: A = A @ A.T + torch.eye(3)*1e-3

In [11]: L = torch.linalg.cholesky(A)

In [12]: out = torch.cholesky_inverse(L)

In [13]: A_paddle = paddle.to_tensor(x)

In [14]: A_paddle = A_paddle @ A_paddle.T + paddle.eye(3)*1e-3
W0528 12:16:00.375560 12983 dygraph_functions.cc:71336] got different data type, run type promotion automatically, this may cause data type been changed.

In [15]: L_paddle = paddle.linalg.cholesky(A_paddle)

In [16]: out_paddle = paddle.cholesky_inverse(L_paddle)

In [17]: An = x @ x.T + np.eye(3)*1e-3

In [18]: on = np.linalg.inv(An)

In [19]: np.testing.assert_allclose(out.numpy(), on, rtol=1e-15)
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
Cell In[19], line 1
----> 1 np.testing.assert_allclose(out.numpy(), on, rtol=1e-15)

    [... skipping hidden 1 frame]

File /usr/lib/python3.8/contextlib.py:75, in ContextDecorator.__call__.<locals>.inner(*args, **kwds)
     72 @wraps(func)
     73 def inner(*args, **kwds):
     74     with self._recreate_cm():
---> 75         return func(*args, **kwds)

File ~/venv38dev/lib/python3.8/site-packages/numpy/testing/_private/utils.py:862, in assert_array_compare(comparison, x, y, err_msg, verbose, header, precision, equal_nan, equal_inf, strict)
    858         err_msg += '\n' + '\n'.join(remarks)
    859         msg = build_err_msg([ox, oy], err_msg,
    860                             verbose=verbose, header=header,
    861                             names=('x', 'y'), precision=precision)
--> 862         raise AssertionError(msg)
    863 except ValueError:
    864     import traceback

AssertionError: 
Not equal to tolerance rtol=1e-15, atol=0

Mismatched elements: 9 / 9 (100%)
Max absolute difference: 4.28335145e-12
Max relative difference: 2.79894493e-11
 x: array([[ 0.237564, -0.120313,  0.138811],
       [-0.120313,  0.138848, -0.083284],
       [ 0.138811, -0.083284,  0.249911]])
 y: array([[ 0.237564, -0.120313,  0.138811],
       [-0.120313,  0.138848, -0.083284],
       [ 0.138811, -0.083284,  0.249911]])

In [20]: np.testing.assert_allclose(out_paddle.numpy(), on, rtol=1e-15)
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
Cell In[20], line 1
----> 1 np.testing.assert_allclose(out_paddle.numpy(), on, rtol=1e-15)

    [... skipping hidden 1 frame]

File /usr/lib/python3.8/contextlib.py:75, in ContextDecorator.__call__.<locals>.inner(*args, **kwds)
     72 @wraps(func)
     73 def inner(*args, **kwds):
     74     with self._recreate_cm():
---> 75         return func(*args, **kwds)

File ~/venv38dev/lib/python3.8/site-packages/numpy/testing/_private/utils.py:862, in assert_array_compare(comparison, x, y, err_msg, verbose, header, precision, equal_nan, equal_inf, strict)
    858         err_msg += '\n' + '\n'.join(remarks)
    859         msg = build_err_msg([ox, oy], err_msg,
    860                             verbose=verbose, header=header,
    861                             names=('x', 'y'), precision=precision)
--> 862         raise AssertionError(msg)
    863 except ValueError:
    864     import traceback

AssertionError: 
Not equal to tolerance rtol=1e-15, atol=0

Mismatched elements: 9 / 9 (100%)
Max absolute difference: 4.28335145e-12
Max relative difference: 2.79894493e-11
 x: array([[ 0.237564, -0.120313,  0.138811],
       [-0.120313,  0.138848, -0.083284],
       [ 0.138811, -0.083284,  0.249911]])
 y: array([[ 0.237564, -0.120313,  0.138811],
       [-0.120313,  0.138848, -0.083284],
       [ 0.138811, -0.083284,  0.249911]])

In [21]: np.testing.assert_allclose(out_paddle.numpy(), out.numpy(), rtol=1e-15)

In [22]: np.testing.assert_allclose(out_paddle.numpy(), out.numpy(), rtol=1e-19)
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
Cell In[22], line 1
----> 1 np.testing.assert_allclose(out_paddle.numpy(), out.numpy(), rtol=1e-19)

    [... skipping hidden 1 frame]

File /usr/lib/python3.8/contextlib.py:75, in ContextDecorator.__call__.<locals>.inner(*args, **kwds)
     72 @wraps(func)
     73 def inner(*args, **kwds):
     74     with self._recreate_cm():
---> 75         return func(*args, **kwds)

File ~/venv38dev/lib/python3.8/site-packages/numpy/testing/_private/utils.py:862, in assert_array_compare(comparison, x, y, err_msg, verbose, header, precision, equal_nan, equal_inf, strict)
    858         err_msg += '\n' + '\n'.join(remarks)
    859         msg = build_err_msg([ox, oy], err_msg,
    860                             verbose=verbose, header=header,
    861                             names=('x', 'y'), precision=precision)
--> 862         raise AssertionError(msg)
    863 except ValueError:
    864     import traceback

AssertionError: 
Not equal to tolerance rtol=1e-19, atol=0

Mismatched elements: 2 / 9 (22.2%)
Max absolute difference: 1.38777878e-17
Max relative difference: 1.15346891e-16
 x: array([[ 0.237564, -0.120313,  0.138811],
       [-0.120313,  0.138848, -0.083284],
       [ 0.138811, -0.083284,  0.249911]])
 y: array([[ 0.237564, -0.120313,  0.138811],
       [-0.120313,  0.138848, -0.083284],
       [ 0.138811, -0.083284,  0.249911]])

```

可以看到，torch 与 numpy ，paddle 与 numpy ，两者 rtol 结果大约在 1e-11 ～

torch 与 paddle 两者 rtol 结果大约在 1e-16 ～